src/components/quad-status/BatteryIcon.vue

```diff
@@ -0,0 +1,127 @@
+<template>
+  <div class="battery-icon">
+    <div class="quad-status-contents">
+      <div
+        class="battery-status"
+        :class="classes"
+        :style="{ width: batteryWidth + '%' }"
+      />
+    </div>
+  </div>
+</template>
+<script>
+const NO_BATTERY_VOLTAGE_MAXIMUM = 1.8; // Maybe is better to add a call to MSP_BATTERY_STATE but is not available for all versions
```

Comment:
Only as suggestion... now that you have refactored it... maybe we can get rid of this, now that the code is clean...

We need to call this MSP:
https://github.com/betaflight/betaflight-configurator/blob/d0d3141b7202004575dfd9f2a162b21dc96fbfb7/src/js/msp/MSPHelper.js#L307-L318

and pass the FC.BATTERY_STATE.batteryState to the component. The value will be:

https://github.com/betaflight/betaflight/blob/025ee87a7aca068e3659fd066b8a9afbed123361/src/main/sensors/battery.h#L84-L90

It can be null or undefined (I'm not too sure) if the firmware is old, in this case we need to use the same code than now.

Reply:
Will grab a look at it 😉 thanks for suggestion

Reply:
Configurator now only supports >= API_VERSION_1_42